presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DetermineJoinDistributionType.java

```diff
@@ -134,7 +135,10 @@ private PlanNode getCostBasedJoin(JoinNode joinNode, Context context)
             }
         }
 
-        if (possibleJoinNodes.stream().anyMatch(result -> result.getCost().hasUnknownComponents()) || possibleJoinNodes.isEmpty()) {
+        PlanNodeStatsEstimate buildSideStats = context.getStatsProvider().getStats(joinNode.getRight());
+        boolean isBuildSideStatsUnknown = buildSideStats.getOutputRowCount() == 0 && buildSideStats.confidenceLevel() == LOW;
+
+        if (isBuildSideStatsUnknown || possibleJoinNodes.stream().anyMatch(result -> result.getCost().hasUnknownComponents()) || possibleJoinNodes.isEmpty()) {
```

Comment:
This will not work. It will enter the logic where there are unknown cost for the possible joins, but actually it is not the case here when we have valid estimations for both sides of the join with one side to be low confident 0.

Reply:
Perhaps make getJoinNodeWithCost to return unknown cost is a better solution